### PR TITLE
Refactor: Do the npm compile as a Build step in docker (Multistep build)

### PR DESCRIPTION
**Pull request by stumpylog**
Based on the work of #19, this pull request updates the workflow to build the front end as part of the Docker build process.

This would supersede #19 as it is based on the paperless-ngx dev branch.

### CODE REVIEW

**Review by qcasey — APPROVED** [review]
Thanks for rebasing this PR for us!

I built this image myself manually, works great. The replacement lines for `compile-frontend.sh` set up a dev env correctly too.

As this is `non-trivial` ideally we can get one more approval glance before merge, but LGTM.

**Review by tribut — APPROVED** [review]

**Comment by Tockra** [comment]
Can somebody explain when this features goes from beta branch to main ?

**Pull request merged**
